src/desktop/krita-colors-to-drawpile.py

```diff
@@ -73,13 +272,16 @@ def print_transformed(section, key, out):
 print_transformed("Colors:View", "ForegroundLink", "BrightText")
 print_transformed("Colors:Button", "BackgroundNormal", "Button")
 print_transformed("Colors:Button", "ForegroundNormal", "ButtonText")
-print_transformed("Colors:Button", "ForegroundNormal", "Dark")
-print_transformed("Colors:Selection", "BackgroundNormal", "Highlight")
-print_transformed("Colors:Selection", "ForegroundNormal", "HighlightedText")
-print_transformed("Colors:Button", "BackgroundNormal", "Light")
+print_transformed("Colors:Window", "BackgroundNormal", "Dark", DARK)
+print_transformed("Colors:Selection", "BackgroundNormal", "Highlight", kc = -1.0)
+print_transformed("Colors:Selection", "ForegroundNormal", "HighlightedText", kc = -1.0)
```

Comment:
The system palettes work this way, so changing the Krita ones doesn’t solve that problem. When Drawpile isn’t focused at all, or there is a modal dialogue that prevents the controls from being interactive, reflecting that state is an important UI affordance.

For the dock widgets, their inactive style should be tied to the focus of the window instead of the focus of themselves or their parent. Since [palettes propagate to non-window children](https://doc.qt.io/qt-5/qwidget.html#palette-prop), I think it should be as simple as just having `MainWindow` manage its palette so it copies the active style to the inactive style when it’s focused, and restores the inactive style when it isn’t.

Reply:
The regular Fusion style doesn't work this way though, so our Fusion colors should behave the same. System styles doing this is fine, if you didn't want this, presumably you'd change your system colors to that effect.

Reply:
I made a mistake initially in hacking in an interpretation of inactive state by guessing at the intent and that’s my fault. To avoid any such future editorialising I decided to just implement the colour effects so the converter was reading the Krita palettes the same way Krita does, but I wasn’t finished yet, so you were looking at something incomplete. A condition was missing, causing effects to be applied even though the colour scheme said not to.

I am having a hard time understanding why it’s relevant to Krita’s palettes what Fusion’s built-in palette does. It’s possible to use system palette with Fusion style, or Fusion palette with system style. Krita itself does not offer Fusion palette. It has a separate style selector so its palettes can be used in system or Fusion style. It also includes Breeze palettes which have a separate inactive selection style. Could you please explain more so I can try to understand?

My own mistaken belief about how tightly coupled the style and palette need to be seems to have been the root cause of why high-contrast mode was not working, too. Only `QWindowsVistaStyle::standardPalette` (sometimes) returns system colours, whereas `QWindowsStyle::standardPalette` just uses hard-coded values inherited from `QStyle::standardPalette`. So that problem is fixed now (in the other branch) by passing `QPalette()` instead of `style->standardPalette()` for the “System” palettes.

Even though it didn’t do anything to solve the high-contrast mode issue, I think it makes sense to still keep the manifest change anyway since it more accurately reflects the reality that Drawpile is not designed to run only on Windows Vista; thoughts?

The dock widgets switching to an inactive palette when the window is still active really isn’t the right thing for this UI paradigm; this was something that I had already on my list of things to fix a long time ago, at least for the layer list. This isn’t a palette choice thing, but rather that the way these controls are intended to be used by default is different than how they have been adapted for use in the Drawpile UI.

Reply:
Krita supports the Fusion, Breeze and Macintosh styles. When set to Fusion, it doesn't change the color on any inactive controls either. So the regular Fusion palette is relevant to Fusion Dark, Krita is relevant to the Krita palettes, but it's the same thing: the inactive colors are the same as the active colors.

I'd be fine with keeping the manifest. We don't intentionally support running Drawpile on Windows Vista, so breaking it isn't of particular concern. If the manifest has no gain otherwise, we can also toss it again to possibly enable someone to hobble along with an operating system they're probably not using voluntarily.

Are they really supposed to be used another way? It seems to me like any list widget has this issue, when you focus a different control they turn inactive. Like in the (old) settings dialog, the list on the left goes inactive when you start messing with preferences on the right. The layer list behaves the same way, which looks pretty okay on my system palettes, since their inactive colors aren't that glaringly different. But overriding it is also fine, I don't think anyone would get confused about the focus state of the docks without the dimming effect.